src/test/archive/archive_node_tests/load_genesis_ledger.ml

```diff
@@ -0,0 +1,112 @@
+open Async
+open Core
+open Mina_automation
+open Mina_automation_fixture.Archive
+
+(**
+  Test file for memory leak detection in archive process.
+  
+  This test is designed to identify potential memory leaks that occur during
+  the archive process, specifically focusing on session cleanup issues with
+  PostgreSQL connections. The memory leak manifests as sessions not being
+  properly cleaned up, leading to an abrupt increase in PostgreSQL memory
+  consumption over time.
+  
+  The test validates that:
+  - Archive operations properly release database sessions
+  - PostgreSQL memory usage remains stable during archive cycles
+*)
+
+type t = Mina_automation_fixture.Archive.before_bootstrap
+
+let postgres_user_name = "postgres"
+
+let check_postgres_memory_increase_is_below_threshold logger threshold
+    time_interval previous_postgres_memory =
+  let%bind current_memory =
+    Utils.get_memory_usage_mb_of_user_process postgres_user_name
+  in
+  let increase = Float.( - ) current_memory !previous_postgres_memory in
+  if Float.( > ) increase threshold then
+    failwith
+      (Printf.sprintf
+         "Postgres memory increased by more than %s MB in the last %s seconds"
```

Comment:
Why fail on this case? What does it indicate if we have a steep in crease in mem usage?

Reply:
It detects a peak of mem usage. I thought it also helpful to test this sudden peak which is unwanted and was one of caqti memory leak observation. I can remove it an focus only on measuring level of memory usage, if you want

Reply:
I think it'd be better if we add a comment there if we keep this :)